android/app/src/main/java/com/vibecode/network/NetworkDiagnosticsModule.kt

```
@@ -0,0 +1,174 @@
+      else -> -1
+    }
+  }
+
+  private fun formatIp(ip: Int): String {
+    return String.format(
+      "%d.%d.%d.%d",
+      ip and 0xff,
+      ip shr 8 and 0xff,
+      ip shr 16 and 0xff,
+      ip shr 24 and 0xff,
+    )
```

detekt (1.23.8)

[warning] String.format(
      "%d.%d.%d.%d",
      ip and 0xff,
      ip shr 8 and 0xff,
      ip shr 16 and 0xff,
      ip shr 24 and 0xff,
    ) uses implicitly default locale for string formatting.

(detekt.potential-bugs.ImplicitDefaultLocale)